server/controllers/userController.js

```diff
@@ -66,6 +70,76 @@ const userController = {
           token
         });
       });
+  },
+  /**
+   * @method updateProfile
+   * @description Allows users to edit their profile
+   * @param {Object} req The request object
+   * @param {Object} res The the response object
+   * @returns {Object} The modified profile details
+   */
+  updateProfile: (req, res) => {
+    multerUploads(req, res, () => {
+      const { currentUser } = req;
+      let image = '';
+      const file = !req.file ? '../../upload/dummy-profile.png' : dataUri(req);
```

Comment:
I'm guessing, you are uploading a default image if the user did not pass an image. Instead, what we should do is have a default image URL in the column, and not make any call to cloudinary if the user did not upload a file.